game-core/src/main/java/games/strategy/triplea/delegate/battle/MustFightBattle.java

```diff
@@ -576,6 +576,7 @@ private static void removeFromDependents(
   @Override
   public void fight(final IDelegateBridge bridge) {
     removeUnitsThatNoLongerExist();
```

Comment:
The pattern of mutating void method calls is unfortunate. The reasoning/logic behind the ordering is not captured, makes it really brittle as re-ordering lines or adding functionality can cause unexpected problems.

It's also not clear as well why the ordering is as it is, we also run into issues as well for reading the functions as the void calls you have to trace down to see what is happening. Much of this perhaps could just as well be inlined and/or made functional so the control flow is captured at a high level. Regardless, it's just kinda sad to see this anti-pattern. There are some interesting tools out there that would perhaps catch this, those tools permute lines and re-run tests to see if any file due to line permutations and then gives you a score of methods that fail highlighting brittlness.